site/content/rfds/0004-plugin-api.md

```diff
@@ -0,0 +1,932 @@
+  format.
+- __Entrypoint__ (`Node`): The command to run the plugin's executable
+  artifact from the command line.
+  - (Child) __On__ (`Node`): An entry for a specific architecture.
+    - (Named Attribute) __Arch__ (`String`): The target triple for the
+      architecture whose entrypoint is being specified.
+    - (Positional Attribute) __Operation__ (`String`): The entrypoint CLI
+      program to run.
+- __License__ (`String`): An [SPDX license expression][spdx_license_expr]
+  specifying all licenses which apply to the plugin. See the
+  [SPDX license list][spdx_license_list] for the full list of known named
+  licenses.
+- __Dependencies__ (`Node`): A node containing all dependency information
+  for the plugin. This is the set of plugins whose outputs the current plugin
+  consumes in the course of its own execution.
+  - (Child) __Plugin__ (`Node`): An individual plugin node, specifying the
+    following attributes:
+    - (Positional Attribute) __Name__ (`String`): The name of the plugin, of
+      the form `<PUBLISHER>/<NAME>`.
+    - (Named Attribute) __Version__ (`String`): The version, specified as a
+      SemVer version constraint of [the forms accepted by Rust's Cargo
+      ecosystem][cargo_semver]. Namely, the following operators are supported
+      (note that this list explicitly _does not_ include
+      [hyphen-ranges][npm_hyphen_ranges] or [x-ranges][npm_x_ranges], both of
+      which are supported in the NPM ecosystem.)
+      - _Caret Requirements_: Permits "SemVer-compatible" updates. Example:
+        `^1.2.3`.
+      - _Tilde Requirements_: Specifies a minimal version with some ability to
+        update. Example: `~1.2.3`.
+      - _Wildcard Requirements_: Permits any version in the place of the
+        wildcard. Example: `1.2.*`
+      - _Comparison Requirements_: Manually specify a version constraint using
+        less than, less than or equal to, equal to, greater than or equal to,
+        or greater than operators. Example: `>= 1.2.3`.
+      - _Multiple Version Requirements_: Permits combining any of the above
+        forms of constraints by separating them with a comma.
+        Example: `>= 1.2.3, < 2.0.0`.
+    - (Named Attribute) __Manifest__ (Optional / `String`): The download
+      location of the __plugin discovery manifest__ for this plugin. Until a
+      plugin registry is established for Hipcheck plugins, this attribute will
+      effectively be required, as otherwise Hipcheck will have no mechanism to
+      discover the location of the download metadata for a plugin.
+
+Given the above definitions, an example plugin manifest may look like:
+
+```kdl
+publisher "mitre"
+name "affiliation"
+version "0.1.0"
+license "Apache-2.0"
+entrypoint {
+  on arch="aarch64-apple-darwin" "./hc-mitre-affiliation"
+  on arch="x86_64-apple-darwin" "./hc-mitre-affiliation"
+  on arch="x86_64-unknown-linux-gnu" "./hc-mitre-affiliation"
+  on arch="x86_64-pc-windows-msvc" "./hc-mitre-affiliation"
```

Comment:
Just confirming that that would be the correct entrypoint even on Windows.

Reply:
Thanks! I do think that we'll likely want to enhance the target triple stuff to permit some "grouping expressions" when granularly listing every target triple is too tedious. Rust has a version of this in the `cfg` system where you can just say "windows" for example and cover _all_ windows targets.